lib/rules/component-defined-inline.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "Ensure component object is defined inline, and not created as a class or named object",
+      category: "Possible Errors",
+      recommended: true
+    },
+    schema: []
+  },
+  create: (context) => {
+    return {
+      'MemberExpression': (node) => {
+        if (node.object.name !== 'angular' && node.property.name !== 'module') {
+          return
+        }
```

Comment:
Do we need to check the argument to module is a literal ? What else could it be ? 

Based on test it looks like it works but I'm pretty confused about the implementation until I open it in AST explorer (i think that is a given for AST stuff at first glance though ...).

By giving each of your LHS a name it might be slightly more sane :
      
    const angularDefinitionType = node.parent.parent.property.name
    const angularComponentBodyType = node.parent.parent.parent.arguments[1].type   
      
    if (angularDefinitionType === 'component' && angularComponentBodyType !== 'ObjectExpression') {
      context.report(node, 'Component not created as in-line anonymous object');
    }